mcodingbot/utils/peps.py

```diff
@@ -48,12 +49,36 @@ async def fetch_pep_info(self, bot: Bot) -> None:
     def get(self, pep_number: int) -> PEPInfo | None:
         return self._peps.get(pep_number)
 
+    def _get_matches_digits(
+        self, query: str, limit: int | None
+    ) -> tuple[Iterable[PEPInfo], int]:
+        """
+        Returns a tuple of (Items found, Amount of items found).
+        """
+        items_iter = (
+            value
+            for key, value in self._peps.items()
+            if str(key).startswith(query)
+        )
+        items = list(itertools.islice(items_iter, limit))
+        return items, len(items)
+
     def search(
         self, query: str, *, limit: int | None = None
     ) -> Iterator[PEPInfo]:
+        yielded = 0
+        if query.isdigit():
+            items, yielded = self._get_matches_digits(query, limit)
+            yield from items
+
         res = fuzzy_search(query, self._pep_map, limit=limit)
         for pep in res:
             if pep_info := self.get(pep[2]):
+                if limit and yielded >= limit:
+                    return
+                if pep_info in items:
+                    continue
+                yielded += 1
                 yield pep_info
```

Comment:
Just had a thought. Would it be better to add a check right after line 72:
```py
if yielded >= limit:
    return
```

That way the fuzzy search is never reached? Then the limit check and the `continue` can be flipped back.

Reply:
Nah, I would rather not put the same check in twice.

Reply:
ended up doing this